build_tools/scripts/download_file.py

```diff
@@ -35,17 +43,26 @@ def main(args):
   if not os.path.isdir(output_dir):
     os.makedirs(output_dir)
 
-  response = requests.get(args.source_url)
-  if response.status_code != 200:
-    raise requests.RequestException(
-        f"Failed to download file with status code {response.status_code}")
+  with urllib.request.urlopen(args.source_url) as response:
```

Comment:
Does this really stream by default? I don't see anything about that in the docs. It looks like for `tarfile` specifically, you're opening in streaming mode, but gzip would still open the whole thing. Can you add some comments about what's going on here?

Reply:
Added the comments.

The streaming behavior of `urlopen` is vaguely mentioned in https://docs.python.org/3/library/http.client.html#httpresponse-objects.

I didn't find an official doc about the streaming mode of gzip, although it looks like that from my test (and it's also natural for gzip to be a stream). There are several sections in https://docs.python.org/3/library/gzip.html#gzip.GzipFile mentioned the support about buffered io.